openedx/core/djangoapps/public_api/courses/tests/test_course_import_export.py

```diff
@@ -0,0 +1,416 @@
+
+    def _outside_tar2(self):
+        """
+        Tarfile with file that extracts to outside directory.
+
+        The path here matches the basename (`self.unsafe_common_dir`), but
+        then "cd's out". E.g. "/usr/../etc" == "/etc", but the naive basename
+        of the first (but not the second) is "/usr"
+
+        Extracting this tarfile in directory <dir> will also put its contents
+        directly in <dir> (rather than <dir/tarname>).
+        """
+        outside_tar = self.unsafe_common_dir / "unsafe_file.tar.gz"
+        with tarfile.open(outside_tar, "w:gz") as tar:
+            tar.addfile(
+                tarfile.TarInfo(str(self.unsafe_common_dir / "../a_file"))
+            )
+
+        return outside_tar
+
+    def test_unsafe_tar(self):
+        """
+        Check that safety measure work.
+
+        This includes:
+            'tarbombs' which include files or symlinks with paths
+        outside or directly in the working directory,
+            'special files' (character device, block device or FIFOs),
+
+        all raise exceptions/400s.
+        """
+
+        def try_tar(tarpath):
+            """ Attempt to tar an unacceptable file """
+            with open(tarpath) as tar:
+                args = {"name": tarpath, "course-data": [tar]}
+                resp = self.client.post(self.url, args)
+            self.assertEquals(resp.status_code, 400)
+            self.assertTrue("suspicious_operation_message" in resp.content)
+
+        try_tar(self._fifo_tar())
+        try_tar(self._symlink_tar())
+        try_tar(self._outside_tar())
+        try_tar(self._outside_tar2())
+        # Check that `ImportStatus` returns the appropriate stage (i.e.,
+        # either 3, indicating all previous steps are completed, or 0,
+        # indicating no upload in progress)
+        resp_status = self.client.get(
+            course_url(
+                'course_import_status_handler',
+                self.course,
+                kwargs={'filename': os.path.split(self.good_tar)[1]}
+            )
+        )
+        import_status = json.loads(resp_status.content)["ImportStatus"]
+        self.assertIn(import_status, (0, 3))
+
+
+@override_settings(CONTENTSTORE=TEST_DATA_CONTENTSTORE)
+class ExportTestCase(CourseTestCase):
+    """q
+    Tests for export_handler.
+    """
+    def setUp(self):
+        """
+        Sets up the test course.
+        """
+        super(ExportTestCase, self).setUp()
+        self.url = course_url('course_import_export_handler', self.course)
+
+    def test_export_html_unsupported(self):
+        """
+        HTML is unsupported
+        """
+        resp = self.client.get(self.url, HTTP_ACCEPT='text/html')
+        self.assertEquals(resp.status_code, 406)
+
+    def test_export_json_unsupported(self):
+        """
+        JSON is unsupported.
+        """
+        resp = self.client.get(self.url, HTTP_ACCEPT='application/json')
+        self.assertEquals(resp.status_code, 200)
```

Comment:
Ah, this is a mismatch from when I moved the code over from the original API, which didn't support json. Will fix